pkg/registry/populator.go

```diff
@@ -119,6 +156,42 @@ func (i *DirectoryPopulator) loadManifests(imagesToAdd []*ImageInput, mode Mode)
 		// same package linearly.
 		var err error
 		var validImagesToAdd []*ImageInput
+		overwriteBundle := ImageInput{}
+
+		// find the image to overwrite if it exists
+		for j, image := range imagesToAdd {
+			for _, channel := range image.bundle.Channels {
+				// get head of the channel
+				bundle, err := i.querier.GetBundleForChannel(context.TODO(), image.bundle.Package, channel)
+				if err != nil {
+					// Assume that if we can not find a bundle for the package and channel that this is a new bundle
+					continue
+				}
+				if bundle != nil && bundle.GetCsvName() == image.bundle.csv.GetName() {
+					// found the overwriting bundle
+					overwriteBundle = *image
+					// remove the overwriting bundle from the images to add since we will add that bundle first
+					imagesToAdd = append(imagesToAdd[:j], imagesToAdd[j+1:]...)
+					break
+				}
+			}
+		}
+
+		// Add the overwriting bundle first
+		if overwriteBundle != (ImageInput{}) {
+			// Remove existing csv
+			err = i.loader.RmCsv(overwriteBundle.bundle.csv.GetName())
```

Comment:
This should only happen when `i.overwrite == true`. there are tests that depend on the standard failure behavior which fail now because this always overwrites.